security/pkg/k8s/tokenreview/k8sauthn.go

```diff
@@ -43,10 +77,9 @@ func ValidateK8sJwt(kubeClient kubernetes.Interface, targetToken, jwtPolicy stri
 			Token: targetToken,
 		},
 	}
-	if jwtPolicy == jwt.PolicyThirdParty {
-		tokenReview.Spec.Audiences = []string{DefaultAudience}
-	} else if jwtPolicy != jwt.PolicyFirstParty {
-		return nil, fmt.Errorf("invalid JWT policy: %v", jwtPolicy)
+	if checkAudience(targetToken) || RequireAudience.Get() {
```

Comment:
The logic here should not be "OR". If RequireAudience is true, token must have audience, otherwise, the validation should fail.

Reply:
Yes - that's what this line does. If RequireAudience is true, the tokenReview.Spec.Audience is set - so the validation will check that the token has the given audience.

If RequireAudience is false - but the token has any audience ( is a 3P token ) - we also verify the audience. 


Note that the original logic may have a security bug - if the policy was set as 'first party' the check is made without audience,
not clear if K8S will reject a request with audience or not. If it does reject - there is no way to migrate a cluster from 1p to 3p. If it accepts - it's a security bug, so 1p policy can't work.